### DISCUSSION TITLE
Custom Smoke / Complex Stoichiometry

### DISCUSSION THREAD

**Discussion by FinleyRae8** [firemodels/fds#13371] [Q&A]
This was initially sent to Thunderhead support but was directed here :)

Hello,

I am doing my uni thesis model using pyrosim, with the topic being the assessment of pre-ignition off-gassing from thermal runaway in lithium-ion battery powered electric vehicles in enclosed carpark environments. At my workplace we use pyrosim to simulate fires, visibility and temperatures in industrial buildings. For my thesis I want to use pyrosim to model a specific off-gas composition to be released and to monitor the propagation of the gas cloud. 

I have tried a lot of different ways to approach this and can't seem to find out how to successfully model this. In terms of the gas composition I have gathered percentages of each element that makes the gas (e.g. hydrogen, carbon dioxide, carbon monoxide, ethylene). All of the elements that make up the gas are on your pre-defined list). Also will just be modelling this in a rectangle box to emulate an underground carpark.

The main parts I am struggling with are:

making sure I am imputing the correct mass fraction (e.g. Hydrogen makes up 30% of the gas so I am putting the value of hydrogen as 0.30 when entering in the mass fraction, which I am pretty sure is wrong),

and making sure I am emitting this custom gas the correct way (have tried through particles, layered, supply and burner surface. Have also tried changing the reactions too.) 

The information I have from past investigations is pretty much the percentage each element makes up of the total gas, the total volume of gas emitted, the rate it is emitted, and the temperature of the thermal runaway phenomena (pre-ignition so no fire just temp of the gas). If there is any more info I would need please instruct me.

I am asking for potentially a guide or method to approach this problem, if possible using your software? Any help or guidance would be much appreciated. If you need any more info from me to help me at all please ask. If this doesn't make sense, please ask and I will try my best. I am not after a super complex way of doing this, the simpler the better! Any help is much appreciated!
attached is an example of how I am inputting the mass fractions.

![ss1](https://github.com/user-attachments/assets/ec2eaf71-b8ec-4c13-8f76-5a03e219fc73)

**Comment by mcgratta** [comment]
The first question to ask is are these really mass fractions or volume fractions. It is more common to report chemical compositions of gas mixtures in terms of volume fractions. 30% hydrogen by mass is a very large volume of hydrogen relative to the other components. 

Now, if all you want to do is track this gas mixture through the car park, you don't need to specifically list all the components. For the resolution you will use for the calculation, the differences in transport properties of the components will not matter. The only thing that will matter is the average molecular weight of this "gas". You can just use a `SPEC` line like this
```
&SPEC ID='my gas`, MW=... /
```
and then specify a mass generation rate of this gas using `SURF` a line applied to a solid boundary or particle. 

Now if you want to burn this gas, you have to decide on what level of detail. I'll leave that for another post.

**Reply by FinleyRae8** [reply]
Thank you for your response! I was getting no Feeback in my results but figured out I actually had a hidden obstruction blocking my surface. I am going to implement your recommended method as an alternative way of modelling what I am after and compare to my initial results as well as results from other literature (even though there is minimal at the moment). If you are interested in how it progresses, I will happily forward results to you :).

**Reply by FinleyRae8** [reply]
I also changed it to % of volume fraction which seemed to fix it too.

**Comment by johodges** [comment]
One of the first decisions you need to make for your thesis is the scope of what you are planning to predict and what you are planning to prescribe. At the scale of dispersion in a car park, you are most likely going to prescribe the battery source term entirely (i.e., the output of a thermal runaway test or analytical calculation is an input to the CFD model) and predict the dispersion over time to evaluate strategies to mitigate an explosion/deflagration hazard. 

If that is the case, I recommend you read my colleagues paper here as a starting point (A CFD based methodology to design an explosion prevention system for Li-ion based battery energy storage system - https://doi.org/10.1016/j.jlp.2023.105038). The team presented a method to use FDS to design an explosion prevention system for an energy storage enclosure. Similar physics at play to the enclosed car park, but much less free air volume.

**Reply by FinleyRae8** [reply]
The scope of my thesis is just to review the gas cloud and how it moves in an eclosed space and analyse the concentration at certain distances and time stamps to determine if occupants egressing the building will be safe! This is to get a better understanding of what safety implications may be needed in future infrastructure surrounding not only EV charging ports, but entire carparks as EVs are becoming more and more common. 

Thank you for recommending your colleagues paper too! this is the first I have seen using FDS as a way to investigate LIBs which is not only interesting but makes me more comfortable doing my simulations through FDS too! 

The main variables I plan on changing will be the composition of the gas to emulate the different battery chemistries, along with different total volumes ect. If you know of a way to set a maximum volume to be emitted that would be great! My current iteration of the model is using a vent and applying a supply surface with a set velocity (average off-gas rate), set surface temperature (av temp of battery under thermal runaway) which injects the specified composition. I know there are more complex/accurate ways to have dynamic temps and emission rates but I am keeping simple for now. Are there any additions or changes you would make to this model to enhance the results?
